### PR TITLE
pulseeffects,easyeffects: 5.0.3 -> 6.0.0

### PR DESCRIPTION
**Pull request by kevincox**
New release, the main feature is updating to GTK4 and significant updates to the internal processing pipelines.

Many dependencies no longer seem to be required, I have manually checked that mentioned plugins are still available.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by cole-h** [comment]
@ofborg eval

Just GitHub being GitHub.

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -19749,10 +19749,12 @@ in
 
   libpulseaudio = libpulseaudio-vanilla;
 
-  pulseeffects-pw = callPackage ../applications/audio/pulseeffects {
-    boost = boost172;
+  easyeffects = callPackage ../applications/audio/easyeffects {
+    glibmm = glibmm_2_68;
   };
 
+  pulseeffects-pw = lib.warn "pulseeffects-pw has been renamed to easyeffects." easyeffects; # Deprecated
```
Comment (jtojnar) [inline pkgs/top-level/all-packages.nix:19756]: This should be moved to `aliases.nix`. Also, I am not sure warnings are supported in top-level attributes (due to nix-env).

Comment (kevincox) [reply pkgs/top-level/all-packages.nix:19756]: Updated.

**Comment by kevincox** [comment]
Result of `nixpkgs-review pr 129564` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>nixos-install-tools</li>
  </ul>
</details>
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>easyeffects (pulseeffects-pw)</li>
  </ul>
</details>

**Comment by jtojnar** [comment]
When I start this, I see the following in the console:

```
(easyeffects:1378817): easyeffects-WARNING **: 02:40:51.314: lv2_wrapper: Could not find the plugin: http://calf.sourceforge.net/plugins/BassEnhancer

(easyeffects:1378817): easyeffects-WARNING **: 02:40:51.326: lv2_wrapper: Could not find the plugin: http://lsp-plug.in/plugins/lv2/compressor_stereo

(easyeffects:1378817): easyeffects-WARNING **: 02:40:51.350: lv2_wrapper: Could not find the plugin: http://calf.sourceforge.net/plugins/Deesser

(easyeffects:1378817): easyeffects-WARNING **: 02:40:51.355: lv2_wrapper: Could not find the plugin: http://lsp-plug.in/plugins/lv2/comp_delay_x2_stereo

(easyeffects:1378817): easyeffects-WARNING **: 02:40:51.366: lv2_wrapper: Could not find the plugin: http://lsp-plug.in/plugins/lv2/para_equalizer_x32_lr

(easyeffects:1378817): easyeffects-WARNING **: 02:40:51.370: lv2_wrapper: Could not find the plugin: http://calf.sourceforge.net/plugins/Exciter

(easyeffects:1378817): easyeffects-WARNING **: 02:40:51.374: lv2_wrapper: Could not find the plugin: http://calf.sourceforge.net/plugins/Filter

(easyeffects:1378817): easyeffects-WARNING **: 02:40:51.379: lv2_wrapper: Could not find the plugin: http://calf.sourceforge.net/plugins/Gate

(easyeffects:1378817): easyeffects-WARNING **: 02:40:51.383: lv2_wrapper: Could not find the plugin: http://calf.sourceforge.net/plugins/Limiter

(easyeffects:1378817): easyeffects-WARNING **: 02:40:51.387: lv2_wrapper: Could not find the plugin: http://lsp-plug.in/plugins/lv2/loud_comp_stereo

(easyeffects:1378817): easyeffects-WARNING **: 02:40:51.395: lv2_wrapper: Could not find the plugin: urn:zamaudio:ZaMaximX2

(easyeffects:1378817): easyeffects-WARNING **: 02:40:51.400: lv2_wrapper: Could not find the plugin: http://calf.sourceforge.net/plugins/MultibandCompressor

(easyeffects:1378817): easyeffects-WARNING **: 02:40:51.404: lv2_wrapper: Could not find the plugin: http://calf.sourceforge.net/plugins/MultibandGate

(easyeffects:1378817): easyeffects-WARNING **: 02:40:51.416: lv2_wrapper: Could not find the plugin: http://calf.sourceforge.net/plugins/Reverb

(easyeffects:1378817): easyeffects-WARNING **: 02:40:51.438: lv2_wrapper: Could not find the plugin: http://calf.sourceforge.net/plugins/StereoTools

(easyeffects:1378817): easyeffects-WARNING **: 02:40:51.462: lv2_wrapper: Could not find the plugin: http://calf.sourceforge.net/plugins/BassEnhancer

(easyeffects:1378817): easyeffects-WARNING **: 02:40:51.467: lv2_wrapper: Could not find the plugin: http://lsp-plug.in/plugins/lv2/compressor_stereo

(easyeffects:1378817): easyeffects-WARNING **: 02:40:51.484: lv2_wrapper: Could not find the plugin: http://calf.sourceforge.net/plugins/Deesser

(easyeffects:1378817): easyeffects-WARNING **: 02:40:51.489: lv2_wrapper: Could not find the plugin: http://lsp-plug.in/plugins/lv2/comp_delay_x2_stereo

(easyeffects:1378817): easyeffects-WARNING **: 02:40:51.500: lv2_wrapper: Could not find the plugin: http://lsp-plug.in/plugins/lv2/para_equalizer_x32_lr

(easyeffects:1378817): easyeffects-WARNING **: 02:40:51.505: lv2_wrapper: Could not find the plugin: http://calf.sourceforge.net/plugins/Exciter

(easyeffects:1378817): easyeffects-WARNING **: 02:40:51.509: lv2_wrapper: Could not find the plugin: http://calf.sourceforge.net/plugins/Filter

(easyeffects:1378817): easyeffects-WARNING **: 02:40:51.513: lv2_wrapper: Could not find the plugin: http://calf.sourceforge.net/plugins/Gate

(easyeffects:1378817): easyeffects-WARNING **: 02:40:51.517: lv2_wrapper: Could not find the plugin: http://calf.sourceforge.net/plugins/Limiter

(easyeffects:1378817): easyeffects-WARNING **: 02:40:51.522: lv2_wrapper: Could not find the plugin: http://lsp-plug.in/plugins/lv2/loud_comp_stereo

(easyeffects:1378817): easyeffects-WARNING **: 02:40:51.526: lv2_wrapper: Could not find the plugin: urn:zamaudio:ZaMaximX2

(easyeffects:1378817): easyeffects-WARNING **: 02:40:51.533: lv2_wrapper: Could not find the plugin: http://calf.sourceforge.net/plugins/MultibandCompressor

(easyeffects:1378817): easyeffects-WARNING **: 02:40:51.537: lv2_wrapper: Could not find the plugin: http://calf.sourceforge.net/plugins/MultibandGate

(easyeffects:1378817): easyeffects-WARNING **: 02:40:51.560: lv2_wrapper: Could not find the plugin: http://calf.sourceforge.net/plugins/Reverb

(easyeffects:1378817): easyeffects-WARNING **: 02:40:51.574: lv2_wrapper: Could not find the plugin: http://calf.sourceforge.net/plugins/StereoTools


```

**Comment by fufexan** [comment]
I can confirm the same output when launching from a terminal. In addition to that, using any plugins for playback results in no sound.

**Comment by kevincox** [comment]
Maybe https://github.com/NixOS/nixpkgs/pull/129564/files#diff-eee1afe4292456d2515930800a69e8e56ae02c4ab6361fe9292c7c7173d59129L107 needs to be restored then? I admit that I didn't try every plugin but thought that if they appeared and could be created it was all good. Would you be able to confirm that fixes the issue?

**Comment by fufexan** [comment]
Do you mean this line?
```nix
caddy1 = throw "caddy 1.x has been removed from nixpkgs, as it's unmaintained: https://github.com/caddyserver/caddy/blob/master/.github/SECURITY.md#supported-versions"; # added 2020-10-02
```

**Comment by kevincox** [comment]
Ah no. I think GitHub links to "deleted" files don't work.

```
  preFixup = ''
    gappsWrapperArgs+=(
      --set LV2_PATH "${lib.makeSearchPath "lib/lv2" lv2Plugins}"
      --set LADSPA_PATH "${lib.makeSearchPath "lib/ladspa" ladspaPlugins}"
    )
  '';
```

I think at least the LV2 is still necessary.

**Pull request merged**